tests/multiple_hooks_modify_params.phpt

```diff
@@ -0,0 +1,17 @@
+--TEST--
+Check if hooks receive modified arguments
+--EXTENSIONS--
+otel_instrumentation
+--FILE--
+<?php
+\OpenTelemetry\Instrumentation\hook(null, 'helloWorld', fn(mixed $object, array $params) => [++$params[0]]);
+\OpenTelemetry\Instrumentation\hook(null, 'helloWorld', fn(mixed $object, array $params) => [++$params[0]]);
```

Comment:
This callback receives the original value (`"a"`) instead of the updated value (`"b"`), we update `execute_data` without updating the `$params` array:
https://github.com/open-telemetry/opentelemetry-php-instrumentation/blob/f2c1a1e87ef540f2d70e2d3bdc990a32f04307ec/otel_observer.c#L165-L183